lib/java_buildpack/container/tomcat/tomcat_geode_store.rb

```diff
@@ -0,0 +1,168 @@
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2017 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'java_buildpack/component/versioned_dependency_component'
+require 'java_buildpack/container'
+require 'java_buildpack/container/tomcat/tomcat_utils'
+require 'java_buildpack/logging/logger_factory'
+
+module JavaBuildpack
+  module Container
+
+    # Encapsulates the detect, compile, and release functionality for Tomcat Redis support.
+    class TomcatGeodeStore < JavaBuildpack::Component::VersionedDependencyComponent
+      include JavaBuildpack::Container
+
+      # (see JavaBuildpack::Component::VersionedDependencyComponent#supports?)
+      def supports?
```

Comment:
Method should be [protected](https://github.com/cloudfoundry/java-buildpack/blob/master/lib/java_buildpack/framework/new_relic_agent.rb#L46-L53)